the-collab-lab/tcl-66-smart-shopping-list
# 23. Delete List Feature

## Description

- `SingleList`'s now have a delete button
-  If the owner of a list that has been shared deletes the list, it deletes it for everyone (and all the nested data)
- If a non-owner deletes a list that has been shared with them, it only deletes it for them (not the nested data)
- All lists have the ability to be deleted
- Users sharedLists array are updated upon deletion, regardless of who deletes

## Related Issue



## Acceptance Criteria

- [x] Create a delete button for lists
- [x] Only the owner of a list can completely delete a list and all its data
- [x] If a user that has had a list shared with them deletes the list, it will only delete for them

## Type of Changes

enhancement

## Updates

closes #51 

### Before

<img width="877" alt="Screenshot 2024-03-29 at 10 05 20 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/4e0b0df1-ab4d-4f6b-86bd-f2ff1380fcb4">

### After

<img width="877" alt="Screenshot 2024-03-29 at 10 05 06 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/6573efa2-396f-4d34-b137-0f22f18cfa93">



## Testing Steps / QA Criteria

1. git pull from main
2. npm checkout an-delete-list
3. navigate to home page
4. click trash can icon to delete list
5. navigate to [firestore](https://console.firebase.google.com/u/3/project/tcl-66-smart-shopping-list/firestore/data/~2Fusers~2Fdaheber@gmail.com) to check if list has been deleted
    - note: you will have to share lists or have a list shared to check functionality of shared lists; you can add a listpath as a ref to someone 's sharedLists to manually share if the share feature hasn't been merged yet
